pulp_ansible/app/tasks/collections.py

```diff
@@ -360,8 +369,10 @@ def _build_url(path_or_url):
                 for item in done:
                     data = parse_metadata(item.result())
 
-                    # v2 uses 'results' as the key while v3 uses 'data'
-                    results = data.get("results") or data.get("data") or [data]
+                    if api_version < 3 or "results" in data.keys():
```

Comment:
Need to check `results` because:
```
(pulp) (pulp) [vagrant@pulp3-source-fedora31 pulp_ansible]$ nc 127.0.0.1 4444
> /home/vagrant/devel/pulp_ansible/pulp_ansible/app/tasks/collections.py(345)_fetch_collections()
-> url = _get_url(page_count, api_version)
(Pdb) api_version
2
(Pdb) remote.url
'https://galaxy.ansible.com/api/v2/collections/testing/k8s_demo_collection'
(Pdb) c




^C
(pulp) (pulp) [vagrant@pulp3-source-fedora31 pulp_ansible]$ nc 127.0.0.1 4444
Ncat: Connection refused.
(pulp) (pulp) [vagrant@pulp3-source-fedora31 pulp_ansible]$ nc 127.0.0.1 4444
> /home/vagrant/devel/pulp_ansible/pulp_ansible/app/tasks/collections.py(345)_fetch_collections()
-> url = _get_url(page_count, api_version)
(Pdb) remote.url
'http://pulp3-source-fedora31.lan/pulp_ansible/galaxy/55ec1549-1a44-49ec-b10f-b96cd27c31da/api/v3/collections/testing/k8s_demo_collection'
(Pdb) api_version
3
```

Reply:
I'm confused about why. The `/api/v2/` and `/api/v3/` is shown in both urls.

Reply:
My takeaway is that the `_get_url()` implementation is not parsing correctly in these cases.

Reply:
`_get_url()` is ok, looking the `TestCase`, we can see it syncs from v2: https://github.com/pulp/pulp_ansible/blob/master/pulp_ansible/tests/functional/api/collection/v3/test_sync.py#L39
but it exposes as v3 https://github.com/pulp/pulp_ansible/blob/master/pulp_ansible/tests/functional/api/collection/v3/test_sync.py#L66

The problem is, we don't expose v3 the same way galaxy does, we always expose as `results`, never as `data`
```
(pulp) [vagrant@pulp3-source-fedora31 pulp_ansible]$ nc 127.0.0.1 4444
> /home/vagrant/devel/pulp_ansible/pulp_ansible/app/tasks/collections.py(374)_fetch_collections()
-> if api_version < 3: 
(Pdb) item.result()
DownloadResult(url='http://pulp3-source-fedora31.lan/pulp_ansible/galaxy/31f91a4e-6fa0-4e44-9d74-467a39f64cc6/api/v3/collections/testing/k8s_demo_collection/versions/', artifact_attributes={
'size': 318, 'sha512': 'f0618dd61223fe837d4b2bddbec40612a129e06d10e3587fc128cd729febef274ca62caf9eb170da40daa12294290412efd787c66e24bffd93c3ff321dacfc75', 'sha384': 'e18d39765ff12f443e452f71
68a0b8fe8a0e00511cdb1f6178a2973263b96beff6b71943014ac376ecaf0b63040c2fb7', 'sha256': '999fddf130260adbe24176aedf19f75b71e56fa4e0b207d7aaa08fa829dc01ab', 'sha224': '46aaf28f9ba5092629e9f90a78
7b12ef5a0fb4e0721c02c047ccf647', 'sha1': '1fab81a787871bd8d3f095360028d91bd437b57f', 'md5': '4655e17a0333c7a38bac259162cca1c9'}, path='/var/lib/pulp/tmp/63412@pulp3-source-fedora31.lan/7e313
7e2-6df8-4d9a-b066-5386c86a7445/tmpc9twrbz0', headers=<CIMultiDictProxy('Date': 'Mon, 31 Aug 2020 19:10:05 GMT', 'Server': 'gunicorn/20.0.4', 'Content-Type': 'application/json', 'Vary': 'Acc
ept', 'Allow': 'GET, HEAD, OPTIONS', 'X-Frame-Options': 'SAMEORIGIN', 'Content-Length': '318')>)
(Pdb) data
{'count': 1, 'next': None, 'previous': None, 'results': [{'version': '0.0.3', 'certification': 'needs_review', 'href': '/pulp_ansible/galaxy/31f91a4e-6fa0-4e44-9d74-467a39f64cc6/api/v3/colle
ctions/testing/k8s_demo_collection/versions/0.0.3/', 'created_at': '2020-08-31T19:10:01.766252Z', 'updated_at': '2020-08-31T19:10:01.766271Z'}]}

```

Reply:
we need https://github.com/ansible/galaxy_ng/blob/master/galaxy_ng/app/api/pagination.py#L46